src/content_mapblock.cpp

```diff
@@ -1513,7 +1513,49 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			v3f pos = intToFloat(p, BS);
 
-			std::vector<aabb3f> boxes = n.getNodeBoxes(nodedef);
+			u8 neighbors = 0;
+
+			// locate possible neighboring nodes to connect to
+			if (f.node_box.type == NODEBOX_CONNECTED) {
+				v3s16 p2 = p;
+
+				p2.Y++;
+				MapNode n2 = data->m_vmanip.getNodeNoEx(blockpos_nodes + p2);
```

Comment:
Here's what I tried: use worldedit to set an enormous array of nodes to connected nodes. Then verify:

* After placing all connections are intact
* After restarting the server & client: all connections are intact

I've tried both methods on 500+ long chains of the ```wall:conduit``` test node I made, and it makes all the connections properly, both right after placing and after restarting server and client.